modules/apps/portal-search/portal-search-web/src/main/resources/com/liferay/portal/search/web/internal/custom/facet/portlet/display/template/dependencies/portlet_display_template_checkbox.ftl

```diff
@@ -82,8 +82,24 @@
 								/>
 
 								<@liferay_aui.script>
-									document.getElementById('${namespace}${customRangeBucketDisplayContext.getBucketText()}').onchange = function() {
-										Liferay.Search.FacetUtil.changeSelection(event);
+									document.getElementById('${namespace}${customRangeBucketDisplayContext.getBucketText()}').onclick = function(event) {
+										if ("${customFacetDisplayContext.getAggregationType()}" == "dateRange") {
+											Liferay.Search.FacetUtil.changeSelection(event);
+										}
+										else {
+											event.preventDefault();
+
+											const customRangeElement = document.getElementById('${namespace}customRange');
+
+											if (customRangeElement.classList.contains('hide')) {
+												customRangeElement.classList.remove('hide');
+											}
+											else {
+												if (Liferay.Search.FacetUtil.isCustomRangeValid(event)) {
+													Liferay.Search.FacetUtil.changeSelection(event);
+												}
+											}
+										}
```

Comment:
The `if` blocks seem strangely nested to me. I'd suggest grouping them by use case so the flow is better. I'm thinking something like:

```
event.preventDefault();

const customRangeElement = document.getElementById('${namespace}customRange');

if ("${customFacetDisplayContext.getAggregationType()}" == "dateRange") {
	Liferay.Search.FacetUtil.changeSelection(event);
}
else if (customRangeElement.classList.contains('hide')) {
	customRangeElement.classList.remove('hide');
}
else if (Liferay.Search.FacetUtil.isCustomRangeValid(event)) {
	Liferay.Search.FacetUtil.changeSelection(event);
}
```

`customRangeElement` could possibly not be used, but it cleans up the blocks a lot so to me that's more worth the readability than any most likely negligible performance hit.

Also if this `${namespace}customRange` is a special-named case, we could make it more apparent by grouping it like:

```
event.preventDefault();

const customRangeElement = document.getElementById('${namespace}customRange');

if (customRangeElement) {
	if (customRangeElement.classList.contains('hide')) {
		customRangeElement.classList.remove('hide');
	} else if (Liferay.Search.FacetUtil.isCustomRangeValid(event)) {
		Liferay.Search.FacetUtil.changeSelection(event);
	}
}
else if ('${customFacetDisplayContext.getAggregationType()}' == 'dateRange') {
	Liferay.Search.FacetUtil.changeSelection(event);
}
```

This logic is slightly different though since originally `if (Liferay.Search.FacetUtil.isCustomRangeValid(event))` would always get checked as a last `else`, so you'll need to check if this should be nested within `if (customRangeElement) {`.

Reply:
Also would this make sense to move this into the util so it'd be a single liner like 
```
document.getElementById('${namespace}${customRangeBucketDisplayContext.getBucketText()}').onclick = function(event) {
    Liferay.Search.FacetUtil.changeRangeSelection(event);
}
```